R/columnchart.R

```diff
@@ -846,10 +866,57 @@ Column <- function(x,
                     xaxis = if (NCOL(chart.matrix) > 1) "x2" else "x", yaxis = "y",
                     data.label.font[[i]], is.stacked, data.label.centered)
 
+        # Create annotations separately for each series
+        # so they can be toggled using the legend 
+        for (j in seq_along(overlay.annotation.list))
+        {
+            a.tmp <- overlay.annotation.list[[j]]
+            a.tmp$threshold <- parseThreshold(a.tmp$threshold)
+            tmp.dat <- getAnnotData(annot.data, a.tmp$data, i, 
+                as.numeric = !grepl("Text", a.tmp$type) && a.tmp$data != "Column Comparisons")
+            ind.sel <- extractSelectedAnnot(tmp.dat, a.tmp$threshold, a.tmp$threstype)
+            if (length(ind.sel) == 0)
+                next
+
+            if (!isTRUE(a.tmp$relative.pos >= 0.0) || !isTRUE(a.tmp$relative.pos <= 1.0))
+                stop(paste0("Relative position of annotation overlay should be ", 
+                "between 0.0 (base of bar) to 1.0 (top of bar)"))
```

Comment:
I've removed this check because it is constrained in the GUI control. Also, there's actually no reason why the relative.pos can't be outside these constraints except that it doesn't look good.